base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
The usefulness of a `symset!` not withstanding, we could also think about permitting `setindex!` operations that maintain symmetry. E.g. `A[:,:] = 0` might be expected to work. Also `A[[CartesianIndex(1,2), CartesianIndex(2,1)]] = 1`, although `symset!` would of course be nicer for this case.